src/vm/frames.cpp

```diff
@@ -1001,6 +1001,84 @@ VOID GCFrame::Pop()
 #endif
 }
 
+#ifndef CROSSGEN_COMPILE
+// GCFrame destructor removes the GCFrame from the current thread's explicit frame list.
+// This prevents issues in functions that have HELPER_METHOD_FRAME_BEGIN / END around 
+// GCPROTECT_BEGIN / END and where the C++ compiler places some local variables over 
+// the stack location of the GCFrame local variable after the variable goes out of scope. 
+GCFrame::~GCFrame()
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+    }
+    CONTRACTL_END;
+
+    if (m_Next != NULL)
+    {
+        GCX_COOP_THREAD_EXISTS(m_pCurThread);
```

Comment:
This macro has on-stack object with destructor. It will make this method slow on Windows x86 because of the prolog/epilog will deal with FS:0 chain.

It may be a good idea to do a manual switch here that does not introduce EH into this method.

Reply:
I've added a commit with such change.